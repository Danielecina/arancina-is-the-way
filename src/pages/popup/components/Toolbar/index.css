.toolbar {
  display: flex;
  width: 100%;
  height: 42px;
  align-items: center;
  justify-content: flex-end;
}

.right {
  margin-left: auto;
}
