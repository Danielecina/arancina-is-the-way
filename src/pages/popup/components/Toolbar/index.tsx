import React, {useCallback, useState} from 'react'
import {Button, Tooltip} from 'antd'
import {useHistory} from 'react-router-dom'
import {useIntl} from 'react-intl'
import {TrophyOutlined, HomeOutlined, MoreOutlined} from '@ant-design/icons'

import './index.css'

const Toolbar: React.FC = () => {
  const [returnToHome, setReturnToHome] = useState(false)
  const history = useHistory()
  const onClick = useCallback(() => {
    if (!returnToHome) {
      history.push('/settings')
      setReturnToHome(true)
      return
    }
    history.push('/')
    setReturnToHome(false)
  }, [returnToHome, history])

  return (
    <header className={'toolbar'}>
      <div className={'left'}>
        <Tooltip
          color={'hsla(0,0%,100%,.2)'}
          placement={'right'}
          title={<TooltipInfo />}
        >
          <Button
            icon={<TrophyOutlined />}
            onClick={() => {}}
            shape={'circle'}
            type={'primary'}
          />
        </Tooltip>
      </div>
      <div className={'right'}>
        <Button
          aria-label={'settings'}
          icon={!returnToHome ? <MoreOutlined /> : <HomeOutlined />}
          onClick={onClick}
          shape={'circle'}
          type={'primary'}
        />
      </div>
    </header>
  )
}

const tooltipStyle = {fontWeight: 900}
export const TooltipInfo: React.FC = () => {
  const {formatMessage} = useIntl()
  return <span style={tooltipStyle}>{formatMessage({id: 'comingSoon'})}</span>
}

export default Toolbar
