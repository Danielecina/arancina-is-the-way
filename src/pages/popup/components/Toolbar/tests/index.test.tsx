import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Toolbar from '..'

describe('Toolbar component', () => {
  test('snapshot', () => {
    const element = renderer.create(<Toolbar />).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('expect to see changes to icons when click contributing link', () => {
    render(<MemoryRouter><Toolbar /></MemoryRouter>)
    const button = screen.getByRole('button', {name: /contributing/i})
    const githubIcon = screen.getByRole('img', {name: /github/i})

    expect(githubIcon).toBeTruthy()
    userEvent.click(button)

    const homeIcon = screen.getByRole('img', {name: /home/i})
    expect(homeIcon).toBeTruthy()
  })
})
