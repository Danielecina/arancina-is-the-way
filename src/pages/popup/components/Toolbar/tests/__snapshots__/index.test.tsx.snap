// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component expect to see correct TooltipInfo 1`] = `
<span
  style={
    Object {
      "fontWeight": 900,
    }
  }
>
  Coming soon
</span>
`;

exports[`Toolbar component snapshot 1`] = `
<header
  className="toolbar"
>
  <div
    className="left"
  >
    <button
      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="button"
    >
      <span
        aria-label="trophy"
        className="anticon anticon-trophy"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="trophy"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M868 160h-92v-40c0-4.4-3.6-8-8-8H256c-4.4 0-8 3.6-8 8v40h-92a44 44 0 00-44 44v148c0 81.7 60 149.6 138.2 162C265.7 630.2 359 721.7 476 734.5v105.2H280c-17.7 0-32 14.3-32 32V904c0 4.4 3.6 8 8 8h512c4.4 0 8-3.6 8-8v-32.3c0-17.7-14.3-32-32-32H548V734.5C665 721.7 758.3 630.2 773.8 514 852 501.6 912 433.7 912 352V204a44 44 0 00-44-44zM184 352V232h64v207.6a91.99 91.99 0 01-64-87.6zm520 128c0 49.1-19.1 95.4-53.9 130.1-34.8 34.8-81 53.9-130.1 53.9h-16c-49.1 0-95.4-19.1-130.1-53.9-34.8-34.8-53.9-81-53.9-130.1V184h384v296zm136-128c0 41-26.9 75.8-64 87.6V232h64v120z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="right"
  >
    <button
      aria-label="contributing"
      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-label="github"
        className="anticon anticon-github"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="github"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
          />
        </svg>
      </span>
    </button>
  </div>
</header>
`;
